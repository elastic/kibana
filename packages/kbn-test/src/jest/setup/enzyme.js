/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
