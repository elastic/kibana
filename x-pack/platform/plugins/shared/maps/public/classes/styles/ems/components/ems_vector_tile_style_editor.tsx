/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React from 'react';

import { EuiFormRow, EuiColorPicker } from '@elastic/eui';
import { i18n } from '@kbn/i18n';

interface Props {
  color: string;
  onColorChange: ({ color }: { color: string }) => void;
}

export function EMSVectorTileStyleEditor({ color, onColorChange }: Props) {
  const onChange = (selectedColor: string) => {
    onColorChange({
      color: selectedColor,
    });
  };

  const colorPickerLabel = i18n.translate(
    'xpack.maps.emsVectorTileStyleEditor.colorBlendPickerLabel',
    {
      defaultMessage: 'Color blend',
    }
  );
  const noColorPlaceholder = i18n.translate(
    'xpack.maps.emsVectorTileStyleEditor.colorBlendPickerPlaceholder',
    {
      defaultMessage: 'No color',
    }
  );
  return (
    <EuiFormRow display="columnCompressed" label={colorPickerLabel}>
      <EuiColorPicker
        compressed
        aria-label={colorPickerLabel}
        color={color}
        onChange={onChange}
        secondaryInputDisplay="top"
        isClearable
        format="hex"
        placeholder={noColorPlaceholder}
        aria-placeholder={noColorPlaceholder}
      />
    </EuiFormRow>
  );
}
