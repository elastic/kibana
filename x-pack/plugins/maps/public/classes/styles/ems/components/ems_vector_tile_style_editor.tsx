/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React from 'react';

import { EuiFormRow, EuiColorPicker } from '@elastic/eui';
import { i18n } from '@kbn/i18n';

interface Props {
  color: string;
  onColorChange: ({ color }: { color: string }) => void;
}

export function EMSVectorTileStyleEditor({ color, onColorChange }: Props) {
  const onChange = (selectedColor: string) => {
    onColorChange({
      color: selectedColor,
    });
  };
  return (
    <EuiFormRow
      display="columnCompressed"
      label={i18n.translate('xpack.maps.emsVectorTileStyleEditor.colorFilterPickerLabel', {
        defaultMessage: 'Color filter',
      })}
    >
      <EuiColorPicker
        compressed
        aria-label="Color"
        color={color}
        onChange={onChange}
        secondaryInputDisplay="top"
        isClearable
        format="hex"
        placeholder="No filter"
        aria-placeholder="No filter"
      />
    </EuiFormRow>
  );
}
