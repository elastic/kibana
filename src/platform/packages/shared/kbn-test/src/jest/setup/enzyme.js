/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

/* eslint-env jest */

/**
 * This is a workaround to fix snapshot serialization of emotion classes when rendering React@18 using `import { render } from 'enzyme'`
 * With React@18 emotion uses `useInsertionEffect` to insert styles into the DOM, which enzyme `render` does not trigger because it is using ReactDomServer.renderToString.
 * With React@17 emotion fell back to sync cb, so it was working with enzyme `render`.
 * Without those styles in DOM the custom snapshot serializer is not able to replace the emotion class names.
 * This workaround ensures a fake emotion style tag is present in the DOM before rendering the component with enzyme making the snapshot serializer work.
 */
function mockEnsureEmotionStyleTag() {
  if (!document.head.querySelector('style[data-emotion]')) {
    const style = document.createElement('style');
    style.setAttribute('data-emotion', 'css');
    document.head.appendChild(style);
  }
}

jest.mock('enzyme', () => {
  const actual = jest.requireActual('enzyme');
  return {
    ...actual,
    render: (node, options) => {
      mockEnsureEmotionStyleTag();
      return actual.render(node, options);
    },
  };
});
