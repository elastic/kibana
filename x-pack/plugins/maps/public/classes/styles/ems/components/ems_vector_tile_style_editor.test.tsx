/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { EMSVectorTileStyleEditor } from './ems_vector_tile_style_editor';

describe('EMSVectorTileStyleEditor', () => {
  test('is rendered', () => {
    const component = shallow(
      <EMSVectorTileStyleEditor color="#4A412A" onColorChange={() => {}} />
    );

    expect(component).toMatchSnapshot();
  });
});
