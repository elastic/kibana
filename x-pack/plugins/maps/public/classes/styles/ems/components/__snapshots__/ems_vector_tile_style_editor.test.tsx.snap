// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EMSVectorTileStyleEditor is rendered 1`] = `
<EuiFormRow
  describedByIds={Array []}
  display="columnCompressed"
  hasChildLabel={true}
  hasEmptyLabelSpace={false}
  label="Color blend"
  labelType="label"
>
  <EuiColorPicker
    aria-label="Color blend"
    aria-placeholder="No color"
    color="#4A412A"
    compressed={true}
    format="hex"
    isClearable={true}
    onChange={[Function]}
    placeholder="No color"
    secondaryInputDisplay="top"
  />
</EuiFormRow>
`;
